### PR TITLE
Don't default enable Tickets or SHAMapV2 when running unit tests

### PR DESCRIPTION
**Pull request by scottschurr**
Both Tickets and SHAMapV2 have been around for a while and don't look like they will be enabled on the network soon.  So they are removed from the supportedAmendments list.  This prevents Env from automatically testing with Tickets or SHAMapV2 enabled, although testing with those features can still be explicitly specified.

Other selected drive-by fixes are included.  See notes on the individual commits.

### CODE REVIEW

**Comment by sublimator** [comment]
>>> don't look like they will be enabled on the network soon

Are you guys still using sqlite for the TX history database? Or SOCI + PostGRES ?

**Comment by seelabs** [comment]
My only feedback has to do with naming.

We have the following functions for dealing with features:
1) `supported_amendments`
2) `with_only_features`
3) `supported_features_except`
4) `makeFeatureBitset`
5) `addFeatures`
6) `removeFeatures`

It's not clear what the difference is between `features` and `amendments`, and the names are becoming the strings of primitives that make up the helper function.

Let me propose the following changes:

1) The term "amendment" means supported feature. A feature may be an amendment or not. So we could rename `supported_amendments` to just `amendments`.

2) Some of the helper function names are verbose: `supported_features_except` and `with_only_features`. What if we removed them entirely and replaced them with primitives. `with_only_features` becomes `makeFeatureBitset`, and `supported_features_except` becomes `removeFeatures(amendments(), ...)`. Using the primitives is about as verbose as using the helper functions:
```
with_only_features
makeFeatureBitset
supported_features_except
removeFeatures(amendments
```

**Comment by scottschurr** [comment]
@seelabs: I agree that the names leave something to be desired.  The one that bugs me the most is `supported_amendments`, when all of the other related names refer to `features`.  My inclination is to stop mentioning amendments and only refer to features in this particular context.

I am in the process of making the naming situation slightly worse.  I'm working on a branch where I've added `supported_features_plus` which makes it easy to run unit tests which require `Tickets`.

FWIW it looks like `makeFeatureBitset`, is only used in Env.h currently.  `addFeatures` and `removeFeatures` are defined in Feature.h but unused.  So at the very least we should be able to consolidate some names.

In summary, yeah, I'm in favor of a naming pass.  But maybe that pass should wait until I'm done with my current bit of thrashing.  Does that work for you?  Thanks.

**Comment by seelabs** [comment]
@scottschurr Yes, doing the renaming in another PR is fine with me.

**Comment by scottschurr** [comment]
Squashed and rebased.  CircleCI failure is known and fixed by https://github.com/ripple/rippled/pull/2205.

**Comment by scottschurr** [comment]
Rebased to 0.80.0-rc1.

**Comment by scottschurr** [comment]
Rebased to 0.80.0-rc2.

**Comment by scottschurr** [comment]
The branch for this passed pull request is now used as the base for pull request #2208.  I'm closing this pull request, since the code will be incorporated when #2208 is merged.
